### PR TITLE
clarify use cases for packages

### PR DESCRIPTION
**Pull request by mirnawong1**
clarifying the use cases for dpeendencies.yml and packages.yml per [slack convo](https://getdbt.slack.com/archives/C50NEBJGG/p1698939808257219) in dbt slack community.

### CODE REVIEW

**Comment by VersusFacit** [comment]
I'm gonna offer a soft approval here. I really think you should wait for Jeremy to sign off on the accuracy.

File: website/snippets/_packages_or_dependencies.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+<Tabs>
+<TabItem value="dependencies" label="When to use dependencies.yml">
+
+`dependencies.yml` is designed for the [dbt Mesh](/guides/best-practices/how-we-mesh/mesh-1-intro) and cross-project reference workflow. Consider using it in the following scenarios:
+
+- Use `dependencies.yml` when you need to set up cross-project references between different dbt projects, especially in a dbt Mesh setup.
+- Use `dependencies.yml` when you want to include both projects and non-private dbt packages in your project's dependencies. 
+  - Private packages are not supported in `dependencies.yml` because it intentionally doesn't support Jinja rendering or conditional configuration. This is to maintain static and predictable configuration, and ensures compatibility with other services, like dbt Cloud.
+- Use `dependencies.yml` for organization and maintainability. It can help maintain your project's organization by allowing you to specify hub packages like `dbt_utils`. This reduces the need for multiple YAML files to manage dependencies.
```
Comment (VersusFacit) [inline website/snippets/_packages_or_dependencies.md]: May want to provide a link to the hub on the word hub for the new reader who won't know what this is

Comment (mirnawong1) [reply website/snippets/_packages_or_dependencies.md]: great call out, thank you @VersusFacit !

**Comment by mirnawong1** [comment]
hey @jtcohen6 , whenever you have a chance -- could you provide a technical review this this pr which clarifies the packages.yml/depencies.yml use case based on that community thread between Mila and us? thank you!

**Comment by mirnawong1** [comment]
followed up for tech review

File: website/snippets/_packages_or_dependencies.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+- "Package" dependencies lets you add source code from someone else's dbt project into your own, like a library. 
+- "Project" dependencies provide a different way to build on top of someone else's work in dbt. Refer to [Project dependencies](/docs/collaborate/govern/project-dependencies) for more info.
+- You can rename your existing `packages.yml` to `dependencies.yml` unless you need to use Jinja within your packages specification. This is necessary if you want to add an environment variable with a Git token in a private Git package specification.
+
+There are some important differences between using a `dependencies.yml` compared to a `packages.yml` file:
+
+<Tabs>
+<TabItem value="dependencies" label="When to use dependencies.yml">
+
+`dependencies.yml` is designed for the [dbt Mesh](/guides/best-practices/how-we-mesh/mesh-1-intro) and cross-project reference workflow. Consider using it in the following scenarios:
```
Comment (mirnawong1) [inline website/snippets/_packages_or_dependencies.md]:
```suggestion
`dependencies.yml` is designed for the [dbt Mesh](/best-practices/how-we-mesh/mesh-1-intro) and cross-project reference workflow. Consider using it in the following scenarios:
```

File: website/snippets/_packages_or_dependencies.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+
+## Use cases
+
+Starting from dbt v1.6, `dependencies.yml` has replaced `packages.yml`. The `dependencies.yml` file can now contain both types of dependencies: "package" and "project" dependencies.
+- "Package" dependencies lets you add source code from someone else's dbt project into your own, like a library. 
+- "Project" dependencies provide a different way to build on top of someone else's work in dbt. Refer to [Project dependencies](/docs/collaborate/govern/project-dependencies) for more info.
+- You can rename your existing `packages.yml` to `dependencies.yml` unless you need to use Jinja within your packages specification. This is necessary if you want to add an environment variable with a Git token in a private Git package specification.
+
+There are some important differences between using a `dependencies.yml` compared to a `packages.yml` file:
```
Comment (jtcohen6) [inline website/snippets/_packages_or_dependencies.md]: The information below gets more at the distinction between `packages` and `projects` (the two ways of taking a dependency on another project), versus the distinction between `packages.yml` and `dependencies.yml`. I think framing it as the difference between the two kinds of files is more confusing.

File: website/snippets/_packages_or_dependencies.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+- Use `dependencies.yml` for organization and maintainability. It can help maintain your project's organization by allowing you to specify [dbt Hub packages](https://hub.getdbt.com/) like `dbt_utils`. This reduces the need for multiple YAML files to manage dependencies.
+
+</TabItem>
+
+<TabItem value="packages" label="When to use packages.yml">
+
+Packages allow you to add source code from someone else's dbt project into your own, like a library. Consider using it in the following scenarios:
+
+- Use `packages.yml` when you want to download dbt packages, such as dbt projects, into your root or parent dbt project. Something to note is that it doesn't contribute to the dbt Mesh workflow.
+- Use `packages.yml` to include packages, including private packages, in your project's dependencies. If you have private packages that you need to reference, `packages.yml` is the way to go.
```
Comment (jtcohen6) [inline website/snippets/_packages_or_dependencies.md:29]: Of all the information in this section, I think this is the one point that needs explaining & emphasizing.

I think we should:
- Link to [the docs](https://docs.getdbt.com/docs/build/packages#git-token-method) on embedding a secret via Jinja `{{ env_var() }}`
- Offer some justification in this vein:

> Embedding a git token via Jinja is a workaround for supporting private packages (repositories) via `git` installation. We are aware that this experience is not ideal, as it requires creating a machine user and distributing a git access token. While it remains the currently supported workflow, we plan to add another way of achieving this goal in the future, without requiring the use of a Jinja-embedded secret environment variable. For that reason, `dependencies.yml` does not support Jinja.

**Review by jtcohen6 — COMMENTED** [review]
@mirnawong1 Sorry this took me so long. Let's talk a bit more about which points really need clarifying / emphasizing here.

**Comment by mirnawong1** [comment]
hey @jtcohen6 - thanks for that feedback and made some changes to fold them in/address them. can you have another look when you have a moment pls? thank you!

**Review by jtcohen6 — APPROVED** [review]
Thank you @mirnawong1! This gets at the message that I want us to be communicating

**Comment by mirnawong1** [comment]
woohoo! amazing and thanks @jtcohen6 ! going to merge this now!

**Pull request merged**